src/bases/createStateful.ts

```diff
@@ -122,34 +129,34 @@ const createStateful: StatefulFactory = createEvented
 					throw new Error(`Already observing state with ID '${observedState.id}'`);
 				}
 				const stateful = this;
-				observedState = {
-					id,
-					observable,
-					subscription: observable
-						.observe(id)
-						.subscribe(
-							(item) => setStatefulState(stateful, item), /* next handler */
-							(err) => {
-								/* TODO: Should we emit an error, instead of throwing? */
-								throw err;
-							}, /* error handler */
-							() => unobserve(stateful)), /* completed handler */
-					handle: {
-						destroy() {
-							const observedState = observedStateMap.get(stateful);
-							if (observedState) {
-								observedState.subscription.unsubscribe();
-								observedStateMap.delete(stateful);
-							}
+				const handle = {
+					destroy() {
+						const observedState = observedStateMap.get(stateful);
+						if (observedState) {
+							observedState.subscription.unsubscribe();
+							observedStateMap.delete(stateful);
 						}
 					}
 				};
-				observedStateMap.set(stateful, observedState);
-				return observedState.handle;
+				const subscription = observable
+					.observe(id)
+					.subscribe(
+						(state) => {
+							queueTask(() => { setStatefulState(stateful, state); });
```

Comment:
Do subscriptions happen in a turn?  I didn't think they did.  So is this needed?